app/jobs/delete_job.rb

```diff
@@ -0,0 +1,9 @@
+class DeleteWorker < ActiveJob::Base
+  queue_as :default
+
+  include Sidekiq::Worker
+
+  def perform(catalogue_delete)
+    catalogue_delete.each{|c| Catalogue.find_by(catalogue_num:c).destroy}

```

Comment:
I tried this and run into the following error:
```ActiveRecord::StatementInvalid: SQLite3::SQLException: near "NULL": syntax error: SELECT "catalogues".* FROM "catalogues" WHERE (catalogue_num in NULL)```

will need help to revise this.